### ISSUE TITLE
[CLOSED] Adding gradient accumulation

### ISSUE THREAD

**Issue by jeswan** [nyu-mll/jiant-v1-legacy#974]
<a href="https://github.com/pruksmhc"><img src="https://avatars3.githubusercontent.com/u/10094008?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [pruksmhc](https://github.com/pruksmhc)**
_Wednesday Dec 04, 2019 at 13:41 GMT_
_Originally opened as https://github.com/nyu-mll/jiant/pull/974_

----

This PR is to address the push towards integrating optimizations around fitting larger batch sizes in smaller GPUs.
The introduction to the method of gradient accumulation is beautifully explained here: 
https://medium.com/huggingface/training-larger-batches-practical-tips-on-1-gpu-multi-gpu-distributed-setups-ec88c3e51255
Phil and I are currently running experiments to make sure this works, however, the default is that accumulation_gradient = 1 (which is the original behavior of jiant without this PR). 

----

_**[pruksmhc](https://github.com/pruksmhc)** included the following code: https://github.com/nyu-mll/jiant/pull/974/commits_
